stylix/mk-target.nix

```
@@ -0,0 +1,73 @@
+{
+  # name used for creation of options
+  name,
+
+  # name used for description of options
+  humanName,
+
+  # condition for automatically enabling this target
+  autoEnable ? true,
+
+  # options to be added under `options.stylix.targets.${name}`
+  extraOptions ? { },
+
+  # list of functions with args representing Stylix options; function results
+  # only merged if all relevant Stylix options are not null
+  configElements ? [ ],
+
+  # function with args representing Stylix options; function result always
+  # merged
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
```

Removing these args and moving the `config` definition from L37-onwards into an impoprted module would be one way to resolve https://github.com/danth/stylix/pull/1130#discussion_r2072680408:

```suggestion
let
  configModule =
    { config, lib, ... }:
    let
      cfg = config.stylix.targets.${name};

      functionArgNames =
        fn:
        lib.pipe fn [
          lib.functionArgs
          builtins.attrNames
        ];

      # ...
    in
    {
      config = lib.mkIf (config.stylix.enable && cfg.enable) (
        lib.mkMerge (
          lib.optional (generalConfig != null) (mkConfig generalConfig)
          ++ map mkConditionalConfig configElements
        )
      );
    };
in
```

```nix
imports = [
  {
    options.stylix.targets.${name}.enable =
      config.lib.stylix.mkEnableTarget humanName autoEnable;
  }
  {
    options.stylix.targets.${name} = extraOptions;
  }
  configModule
]
```